learn-arai/app/hooks/useAuth.ts

```diff
@@ -0,0 +1,80 @@
+'use client';
+
+// this react hook responsible for checking if user is logged in.
+import { useEffect, useState } from 'react';
+
+import { useLocalStorage } from './useLocalStorage';
+import { useUser } from './useUser';
+
+export const useAuth = () => {
+    const { addUser, removeUser, user } = useUser();
+    const { getItem } = useLocalStorage();
+    const [isAuthenticated, setIsAuthenticated] = useState<boolean>();
+
+    useEffect(() => {
+        checkSession();
+    });
+
+    const signIn = async (credentials: FormData) => {
+        const response = await fetch(
+            `${process.env.NEXT_PUBLIC_BACKEND_URL}/auth/sign-in`,
+            {
+                method: 'POST',
+                body: credentials,
+                credentials: 'include',
+            }
+        );
+
+        const data = await response.json();
+        const message = data.message;
+        const status = data.status;
+
+        if (status == 'success') {
+            const email = credentials.get('email')!.toString();
+            addUser({ email: email });
+        } else {
+            removeUser();
+        }
+
+        return { message, status };
+    };
+
+    const signOut = () => {
+        removeUser();
+    };
+
+    const checkSession = async () => {
+        const currentPath = window.location.pathname;
+        const isUserEmpty = getItem('user');
+        if (!isUserEmpty && currentPath != '/login') {
+            window.location.href = `/login?redirect=${encodeURIComponent(currentPath)}`;
+            return;
+        }
```

Comment:
This will force you to login, even though you just visit landing page (don't need to login) because `checkSession()` always run when page changed / update.